torch/csrc/autograd/variable.h

```diff
@@ -665,14 +607,10 @@ inline std::shared_ptr<Function> Variable::try_get_grad_accumulator() const {
 }
 
 inline Variable Variable::detach() const {
-  auto var = make_variable_view(*this, get()->data_, /*is_differentiable=*/false, /*allow_tensor_metadata_change=*/false, Edge());
+  auto var = make_variable_view(*this, *this, /*is_differentiable=*/false, /*allow_tensor_metadata_change=*/false, Edge());
```

Comment:
is this right?  It looks pretty strange, but I guess it could be.

Reply:
I think this is right - `*this` points to the same data as the original `get()->data_`, and since `make_variable_view()` makes a shallow-copy of the second parameter `data`, the semantics will be the same before / after the change.